File: fsharp/multiples-of-3-or-5/solution.fs
// https://www.codewars.com/kata/514b92a657cdc65150000006

let triOrQuin = fun n -> n % 3 = 0 || n % 5 = 0

let solve n =
  [0 .. n-1]
    |> List.filter triOrQuin
    |> List.sum

File: haskell/multiples-of-3-or-5/solution.hs
-- https://www.codewars.com/kata/514b92a657cdc65150000006

module MultiplesOf3And5 where

solution :: Integer -> Integer
solution a = sum [b | b <- [1..a-1], b `mod` 3 == 0 || b `mod` 5 == 0]

